### PR TITLE
Generalize centaur test specs, fixed floatingTags for cromwell-2094

### PR DESCRIPTION
**Pull request by cjllanwarne**

### CODE REVIEW

File: src/main/scala/centaur/test/Test.scala [hunk 1]
```diff
@@ -107,43 +107,53 @@ object Operations {
...
+        case None => CentaurCromwellClient.metadata(submittedWorkflow)
       }
     }
   }
 
   /**
     * Verify that none of the calls within the workflow are cached.
     */
-  def validateNoCacheHits(metadata: WorkflowMetadata, workflowName: String): Test[Unit] = {
+  def validateCacheHitField(metadata: WorkflowMetadata, workflowName: String, blacklistedValue: String): Test[Unit] = {
```
Comment (mcovarr) [inline src/main/scala/centaur/test/Test.scala]: "blacklistedValue" doesn't seem like the right term when looking for a cache hit

File: src/main/scala/centaur/test/Test.scala [hunk 2]
```diff
@@ -107,43 +107,53 @@ object Operations {
...
+            validateMetadata(submittedWorkflow, expectedMetadata, cacheHitUUID)
+          }
+          // Nothing to wait for, so just return the first metadata we get back:
+        case None => CentaurCromwellClient.metadata(submittedWorkflow)
       }
     }
   }
 
   /**
     * Verify that none of the calls within the workflow are cached.
```
Comment (mcovarr) [inline src/main/scala/centaur/test/Test.scala]: comment doesn't seem right anymore

**Comment by mcovarr** [comment]
:+1:

[Approved with PullApprove](https://pullapprove.com/broadinstitute/centaur/pull-request/195/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=broadinstitute/centaur)

File: src/main/resources/standardTestCases/backendWithNoDocker.test [hunk 3]
```diff
@@ -1,5 +1,6 @@
 name: backendWithNoDocker
 backends: [LocalNoDocker]
+testFormat: runtwiceexpectingcallcaching
```
Comment (Horneth) [inline src/main/resources/standardTestCases/backendWithNoDocker.test:3]: Can we camel case this or whichever format you prefer ?

File: src/main/scala/centaur/test/Test.scala [hunk 4]
```diff
@@ -107,43 +107,53 @@ object Operations {
...
-        eventually(OffsetDateTime.now(), CentaurConfig.metadataConsistencyTimeout) {
-          validateMetadata(workflow, expectedMetadata, cacheHitUUID)
-        }
+      override def run: Try[WorkflowMetadata] = workflowSpec.metadataExpectation match {
+        case Some(expectedMetadata) =>
+          eventually(OffsetDateTime.now(), CentaurConfig.metadataConsistencyTimeout) {
+            validateMetadata(submittedWorkflow, expectedMetadata, cacheHitUUID)
+          }
+          // Nothing to wait for, so just return the first metadata we get back:
+        case None => CentaurCromwellClient.metadata(submittedWorkflow)
```
Comment (Horneth) [inline src/main/scala/centaur/test/Test.scala:133]: Do we ever want a test where there is no metadata expectation ? Not even Success / Failure ? Or are they checked somewhere else ?

File: src/main/scala/centaur/test/Test.scala [hunk 5]
```diff
@@ -107,43 +107,53 @@ object Operations {
...
     * Verify that none of the calls within the workflow are cached.
     */
-  def validateNoCacheHits(metadata: WorkflowMetadata, workflowName: String): Test[Unit] = {
+  def validateCacheHitField(metadata: WorkflowMetadata, workflowName: String, blacklistedValue: String): Test[Unit] = {
     new Test[Unit] {
       override def run: Try[Unit] = {
-        val cacheHits = metadata.value filter {
-          case (k, JsString(v)) if k.contains("callCaching.result") => v.contains("Cache Hit")
-          case _ => false
+        val cacheHits = metadata.value collect {
```
Comment (Horneth) [inline src/main/scala/centaur/test/Test.scala:144]: this variable is really `cacheResults` now no ?

**Review by Horneth — APPROVED** [review]

**Comment by Horneth** [comment]
👍 

[Approved with PullApprove](https://pullapprove.com/broadinstitute/centaur/pull-request/195/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=broadinstitute/centaur)

File: src/main/scala/centaur/test/Test.scala (hunk 4 above)
Comment (cjllanwarne) [reply src/main/scala/centaur/test/Test.scala:133]: This is the `validateMetadata` `Test` which now returns the validated metadata. 

I guess it's up to the `TestFormat` when to run this but right now they all run it after checking for the workflow completion (which IIRC doesn't use metadata).

File: src/main/scala/centaur/test/Test.scala (hunk 5 above)
Comment (cjllanwarne) [reply src/main/scala/centaur/test/Test.scala:144]: Yeah, good catch

**Comment by cjllanwarne** [comment]
Your comments are addressed in my other PR. I'll close this version

**Pull request closed without merging**